src/main/java/by/yahor_kulesh/dao/TicketDAO.java

```diff
@@ -1,122 +1,63 @@
-    private ResultSet rs = null;
 
-    public void insert(Ticket ticket){
-        try(
-                Connection con = ConnectionConfig.getConnection();
-                PreparedStatement stat = con.prepareStatement("INSERT INTO ticket(id,user_id,ticket_type,creation_date) VALUES (?,?,CAST(? AS ticket_type),?)")
-        ){
-            prepareTicketForStatement(ticket, stat);
-            stat.execute();
-        }catch(SQLException e){
-            System.err.println(e.getMessage());
+    public void saveTicket(TicketEntity ticket){
+        try(Session session = SessionFactoryProvider.getSessionFactory().openSession()){
+            Transaction tx = session.beginTransaction();
+            session.persist(ticket);
+            tx.commit();
+        } catch(Exception e) {
+            System.err.println("Error during saving ticket: " + e.getMessage());
         }
     }
 
-    public Ticket getById(UUID id) {
-        try{
-            try(
-                    Connection con = ConnectionConfig.getConnection();
-                    PreparedStatement stat = con.prepareStatement("SELECT * FROM ticket where id=?")
-            ) {
-                stat.setObject(1, id);
-                rs = stat.executeQuery();
-                if(rs.next()){
-                    return getTicketFromDB();
-                }else return null;
-            } finally {
-                if(rs != null) {
-                    rs.close();
-                }
-            }
-        } catch(SQLException e) {
-            System.err.println(e.getMessage());
+    public TicketEntity getTicketById(UUID id){
+        try(Session session = SessionFactoryProvider.getSessionFactory().openSession()){
+            return session.get(TicketEntity.class, id);
+        } catch(Exception e) {
+            System.err.println("Error during getting ticket: " + e.getMessage());
             return null;
         }
     }
 
-    public ObjectArray getByUserId(UUID id) {
-        ObjectArray tickets = new ObjectArray();
-        try{
-            try(
-                    Connection con = ConnectionConfig.getConnection();
-                    PreparedStatement stat = con.prepareStatement("SELECT * FROM ticket where user_id=?")
-            ) {
-                stat.setObject(1, id);
-                rs = stat.executeQuery();
-                while(rs.next()) {
-                    tickets.add(getTicketFromDB());
-                }
-                return tickets;
-            } finally {
-                if(rs != null) {
-                    rs.close();
-                }
-            }
-        } catch(SQLException e) {
-            System.err.println(e.getMessage());
+    public List<TicketEntity> getTicketByUserId(UUID id) {
+        try(Session session = SessionFactoryProvider.getSessionFactory().openSession()){
+            SelectionQuery<TicketEntity> q = session.createSelectionQuery("SELECT t FROM TicketEntity t WHERE t.user.id = :id", TicketEntity.class);
```

Comment:
the same:
` SelectionQuery<TicketEntity> query ` or ``SelectionQuery<TicketEntity> selectionQuery `
instead of 
` SelectionQuery<TicketEntity> q`